sklearn/linear_model/_ridge.py

```diff
@@ -1416,23 +1414,24 @@ def fit(self, X, y, sample_weight=None):
         y : array-like of shape (n_samples,) or (n_samples, n_targets)
             Target values. Will be cast to float64 if necessary
 
-        sample_weight : float or array-like of shape [n_samples]
+        sample_weight : float or array-like of shape (n_samples,), default=None
             Sample weight
 
         Returns
         -------
         self : object
         """
-        X, y = check_X_y(X, y, ['csr', 'csc', 'coo'],
-                         dtype=[np.float64],
-                         multi_output=True, y_numeric=True)
-
         if np.any(self.alphas <= 0):
             raise ValueError(
                 "alphas must be positive. Got {} containing some "
                 "negative or null value instead.".format(self.alphas))
 
-        sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+        X, y = check_X_y(X, y, ['csr', 'csc', 'coo'], dtype=[np.float64],
+                         multi_output=True, y_numeric=True)
```

Comment:
What's the motivation for moving this here? I would revert to avoid cosmetic changes. In this case it would allow to fail early for wrong alphas, but that's shouldn't be a very common occurrence to optimize for.

Reply:
`_check_sample_weight` uses `X.dtype`, therefore, as in most other places, I'd like to first check `X` and `y` and then sample_weight.

Reply:
Yes, but both `check_X_y` and `_check_sample_weight` were already used here before in that order. At least that's what the diff on Github says...  All this does is to skip the sample weight test `if sample_weight is not None` and so I wondering why we needed to move it around.

Reply:
I think, my motivation was to have both checks together. So either move `_check_sample_weight` up or `check_X_y`down. As `if np.any(self.alpha <= 0)` seems to be the cheaper check, I decided for the latter. I'll revert that and do the former.